### ISSUE TITLE
Support iPhones/iOS

### ISSUE THREAD

**Issue by corvardt** [bradthomasbrown/dizzyhavoc-web#25]
Right now it seems like island-based features aren't working on my iOS device (I've seen people having the same issue), can't connect to faucet/ old bridge with metamask/rabby app
Current bridge chain selectors are unresponsive on metamask/rabby/safari/brave/chrome

Edit:
Device: iPhone 15 pro
iOS Version: 17.2.1

**Comment by bradthomasbrown** [comment]
i think i narrowed it down to [Promise.withResolvers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers) not being supported until iOS/Safari 17.4, which was only released less than a month ago

i updated a lot of dependencies to work around that, let me know if there are issues.

i was able to use the faucet on an iPhone 15 pro, iOS version 17.3.1

**Comment by corvardt** [comment]
all islands working as intended now, nice catch

only thing i've noticed is a duplicate of metamask in provider select screen on MM app browser:
![image](https://github.com/bradbrown-llc/dizzyhavoc-web/assets/124924375/0f4c706d-da2d-4bb8-80db-3cdf52d9e905)

**Comment by bradthomasbrown** [comment]
the duplicate was because metamask injects an EIP6963 provider, but also connects using metamask SDK.
there was a bug in the metamask SDK image source, which should now be fixed so metamask SDK's provider's icon loads correctly.
metamask's SDK doesn't give an EIP5749/EIP6963 UUID, so one was being generated randomly and technically the app is correct in that these are two different providers according to standards.
i added a check before trying to load metamask SDK so if there's already a provider by the name "MetaMask" in the list while its loading, to skip the SDK.

should only be one now

**Comment by corvardt** [comment]
Didn't know multiple instances of metamask running in parallel was a thing, tho they connected to the same address

I can confirm, adding a check for provider with same name fixed the issue

**Comment by bradthomasbrown** [comment]
closing for now, since issues seem to be resolved
